packages/ui/src/Table/Table.tsx

```diff
@@ -0,0 +1,255 @@
+import { DataTestProp } from '@hazelcast/helpers'
+import React, { ReactElement, useEffect } from 'react'
+import cn from 'classnames'
+import {
+  useTable,
+  usePagination,
+  TableOptions,
+  useSortBy,
+  Row as RowType,
+  Cell as CellType,
+  useAsyncDebounce,
+  useResizeColumns,
+  ColumnInterfaceBasedOnValue,
+  useFlexLayout,
+} from 'react-table'
+
+import { Pagination } from '../Pagination'
+import { Cell, CellProps } from './Cell'
+import { EnhancedCellRenderer, EnhancedHeaderFooterRenderer } from './EnhancedRenderers'
+import { Header } from './Header'
+import { Row } from './Row'
+
+import styles from './Table.module.scss'
+
+/**
+ * `defaultPageSize` should be one of the values in `pageSizeOptions`.
+ * If it's not it won't break anything but it is a bad UX
+ * because once user selects different page size he won't be able to select
+ * `defaultPageSize` again.
+ */
+type ExtendedPaginationProps = {
+  defaultPageSize?: number
+  pageSizeOptions?: number[]
+  hidePagination?: boolean
+}
+
+export type FetchDataProps = {
+  pageIndex: number
+  pageSize: number
+}
+
+// When using manual pagination always provide `fetchData` function and our own pageCount
+export type ControlledPaginationProps = {
+  manualPagination?: boolean
+  fetchData?: (fetchDataProps: FetchDataProps) => void
+}
+
+type CustomTableProps<D extends object> = {
+  /**
+   * When using `onRowClick` it's a good practice to also make one of the cells in the table interactive
+   * by providing a button, link or something else.
+   * An example can be found in `ClickableRowsWithNameLink` story in Table.stories.tsx
+   */
+  onRowClick?: (rowInfo: RowType<D>) => void
+  // Custom props getter for Cell
+  getCustomCellProps?: (cellInfo: CellType<D>) => CellProps
+} & DataTestProp
+
+type TableProps<D extends object> = TableOptions<D> & ExtendedPaginationProps & ControlledPaginationProps & CustomTableProps<D>
+
+const column = {
+  // When using the useFlexLayout:
+  minWidth: 30, // minWidth is only used as a limit for resizing
+  width: 150, // width is used for both the flex-basis and flex-grow
+  maxWidth: 200, // maxWidth is only used as a limit for resizing
+}
+
+// Inspiration here: https://react-table.tanstack.com/docs/examples/basic
+export function Table<D extends object>({
```

Comment:
nit: should it be an arrow function for consistency?
BTW, should we export it from `src/index.tsx`?

Reply:
Yes, and yes :+1:

Reply:
Changed here: https://github.com/hazelcast/frontend-shared/pull/127/commits/8dc56f76951dcc6380237547b05fb5cfc24fe457